invesalius/gui/frame.py

```diff
@@ -412,12 +412,31 @@ def _UpdateViewerFocus(self, orientation):
     def CloseProject(self):
         Publisher.sendMessage('Close Project')
 
+    def ExitDialog(self):
+        msg = _("Are you sure you want to exit?")
+        if sys.platform == 'darwin':
+            dialog = wx.RichMessageDialog(None, "", msg, wx.ICON_QUESTION | wx.YES_NO | wx.NO_DEFAULT)
+            dialog.ShowCheckBox("Store session", True)
+        else:
+            dialog = wx.RichMessageDialog(None, msg, "Invesalius 3", wx.ICON_QUESTION | wx.YES_NO | wx.NO_DEFAULT)
+            dialog.ShowCheckBox("Store session", True)
+
+        answer = dialog.ShowModal()
+        
+        if not dialog.IsCheckBoxChecked() and answer == wx.ID_YES:
+            session = ses.Session()
+            self.CloseProject()
+            session.DeleteStateFile()
```

Comment:
For some reason, this is not deleting the state file. Check my comment in the conversation tab about the unwanted behavior.

Reply:
Just looked at it, it seems to be an issue with the order in which the functions are called. I'll fix it asap.

Reply:
The unwanted behaviour is now removed. Additionally, for semantic and functional reasons the session exiting (and subsequent state file deleting) process is moved backed to session.py.